services/ui-src/src/components/reports/OverlayModalPage.tsx

```diff
@@ -50,6 +56,12 @@ export const OverlayModalPage = ({
     }
   }, [report]);
 
+  useEffect(() => {
+    //only update if we're using objectiveCard Templates
+    if (objectiveCards)
+      modalForm = objectiveCards[selectedEntityIndex].modalForm;
+  }, [selectedEntityIndex]);
+
```

Comment:
If you went with the above comment you can swap this to 
```
  useEffect(() => {
    getModalForm();
  }, [selectedStepEntity]);
```

Reply:
this useEffect ended up not being needed once i implemented the function above, so its been removed